File: source-connector/src/main/java/sourceconnector/domain/connect/FileProcessingResult.java
package sourceconnector.domain.connect;

import lombok.Getter;
import lombok.Setter;

@Getter
public class FileProcessingResult {
  @Setter
  private int totalCount = 0;
  private int successCount = 0;
  private int failureCount = 0;
  private int skippedCount = 0;

  public void addSuccessCount() {
    successCount++;
  }

  public void addFailureCount() {
    failureCount++;
  }

  public void addSkippedCount() {
    skippedCount++;
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/connect/FileProcessingResult.kt
package sourceconnector.domain.connect

import offsetmanager.domain.file.FileKey

class FileProcessingResult(
  val totalCount: Int
) {
  var successCount: Int = 0
  val failedFileKeys: MutableList<FileKey> = mutableListOf()
  var skippedCount: Int = 0

  fun addSuccessCount() {
    successCount++
  }

  fun addFailure(fileKey: FileKey) {
    failedFileKeys.add(fileKey)
  }

  fun addSkippedCount() {
    skippedCount++
  }
}
